addOns/exim/src/main/java/org/zaproxy/addon/exim/pcap/PcapImporter.java

```diff
@@ -33,19 +51,93 @@ public PcapImporter(File file) {
 
     public PcapImporter(File file, ProgressPaneListener listener) {
         this.progressListener = listener;
-        this.success = importPcapFile(file);
+        importPcapFile(file);
+    }
+
+    private void importPcapFile(File file) {
+        List<HttpMessage> messages = null;
+
+        try {
+            messages = getHttpMessages(file);
+        } catch (IOException e) {
+            LOGGER.warn("Failed to read Pcap file: {}\n{}", file.getAbsolutePath(), e.getMessage());
+            success = false;
+            completed();
+            return;
+        }
+
+        int count = 0;
+        for (HttpMessage msg : messages) {
+            if (msg == null) {
+                continue;
+            }
```

Comment:
Should the counter also be incremented here? Otherwise if a null actually gets hit in the end it'll show like 9/10 won't it?

Reply:
> Should the counter also be incremented here? Otherwise if a null actually gets hit in the end it'll show like 9/10 won't it?

Good question. It's written in the same way as in the Har importer's processMessages function. Difference with HarImporter is that for the pcap importer I do not set the total tasks to be done as I can not predict that without first parsing the entire http flows from the pcap. What I can do however, is set the total tasks as `messages.size()` after parsing and then indeed increment when a null is hit.

Reply:
Good question. It's written in the same way as in the Har importer's processMessages function. Difference with HarImporter is that for the pcap importer I do not set the total tasks to be done as I can not predict that without first parsing the entire http flows from the pcap. What I can do however, is set the total tasks as messages.size() after parsing and then indeed increment when a null is hit.

Reply:
This had some minor implications (added a message constant when updating progress on a null message and changed the constructor), but I can make the same change to the HarImporter which has likely the same issue if you want me to

Reply:
In the HAR case the number of messages is calculated beforehand, so they should match.